api/storage_proxy.cc

```diff
@@ -66,6 +66,19 @@ future<V> two_dimensional_map_reduce(distributed<service::storage_proxy>& d,
     }, reducer, initial_value);
 }
 
+/**
+ * A helper function to sum values on an a distributed object that
+ * has a get_stats method.
+ *
+ */
+template<class T, class F, class V>
+future<json::json_return_type>  sum_stats(distributed<T>& d, V F::*f) {
+    return d.map_reduce0([f](const T& p) {return p.get_stats().*f;}, 0,
+            std::plus<V>()).then([](V val) {
+        return make_ready_future<json::json_return_type>(val);
+    });
+}
+
```

Comment:
Suggest to move these templates to api/statistics.hh (and only include from api/storage_proxy.cc). These helpers are quite generic, and can be used from other places too.

Reply:
Since no one used it outside storage-proxy.cc, I did not see a use case to extract it. Is that for future inclusion of api/statistics.hh in some other code?

Reply:
In other api/*.cc files that report statistics.